src/authorizations/users.rs

```diff
@@ -1,31 +1,76 @@
-use crate::config::Config;
+use reqwest::blocking::Client as reqwest_client;
+use serde::Deserialize;
 
-pub fn is_authorized(id: &str, config: &Config) -> bool {
-    config.authorized_users.contains(id)
+// Checks for permissions in https://github.com/rust-lang/team/
+#[derive(Debug, Deserialize, Clone)]
+struct TeamResponse {
+    discord_ids: Vec<usize>,
+}
+
+#[derive(Debug)]
+struct TeamClient {
+    client: reqwest_client,
+}
+
+impl TeamClient {
+    pub fn new() -> Self {
+        let team_client = reqwest_client::new();
+
+        TeamClient {
+            client: team_client,
+        }
+    }
+}
+
+pub fn is_authorized(id: &str) -> bool {
+    let authorization_info = team_info();
+    discord_id_in_list(id, authorization_info)
+}
+
+fn discord_id_in_list(id: &str, team_response: TeamResponse) -> bool {
+    team_response
+        .discord_ids
+        .contains(&id.parse::<usize>().unwrap())
+}
+
+fn team_info() -> TeamResponse {
+    get_team_info().unwrap()
+}
+
+fn get_team_info() -> Result<TeamResponse, reqwest::Error> {
+    let team_client = TeamClient::new();
+
+    let team_request = team_client.client.get(
+        &String::from("https://team-api.infra.rust-lang.org/v1/permissions/crates_io_ops_bot.staging_crates_io.json")
+    );
+
+    let team_response = team_request.send().and_then(|res| res.error_for_status())?;
+
+    let response_text = team_response.text().unwrap();
+
+    let team_json: TeamResponse = serde_json::from_str(&response_text).unwrap();
```

Comment:
```suggestion
    let team_json: TeamResponse = team_request
        .send()?
        .error_for_status()?
        .json()?;
```

Also, note that the JSON decoding here could fail, for example if GitHub Pages returns a 5xx error.